x/nexus/keeper/msg_server.go

```diff
@@ -111,43 +113,104 @@ func (s msgServer) DeregisterChainMaintainer(c context.Context, req *types.Dereg
 
 func (s msgServer) ActivateChain(c context.Context, req *types.ActivateChainRequest) (*types.ActivateChainResponse, error) {
 	ctx := sdk.UnwrapSDKContext(c)
-	for _, chainStr := range req.Chains {
-		chain, ok := s.GetChain(ctx, chainStr)
-		if !ok {
-			return nil, fmt.Errorf("%s is not a registered chain", chainStr)
+	if strings.ToLower(req.Chains[0]) == "all" {
+		for _, chain := range s.GetChains(ctx) {
+			s.activateChain(ctx, chain)
 		}
+	} else {
+		for _, chainStr := range req.Chains {
+			chain, ok := s.GetChain(ctx, chainStr)
+			if !ok {
+				return nil, fmt.Errorf("%s is not a registered chain", chainStr)
```

Comment:
what if the other chains are registered? Should we really error out for all of them?

Reply:
I prefer error out and resend it with valid chains, but I don't have strong opinion on this